File: Data Structures/array.py
class myArray:
	def __init__(self):
		self.length = 0
		self.data = {}

	def get(self, index):
		return self.data[index] if self.length > 0 else None

	def printArray(self):
		return list(self.data.values())

	def push(self, value):
		self.data[self.length] = value
		self.length += 1
		return self

	def pop(self):
		if (self.length == 0):
			return "Nothing to pop"
		else:
			popped = self.data[self.length-1]
			del self.data[self.length-1]
			self.length -= 1
			return popped

	def shift(self):
		if (self.length == 0):
			return "Nothing to shift"
		else:
			popped = self.data[0]
			for i in range(self.length-1):
				self.data[i] = self.data[i+1]
			del self.data[self.length-1]
			self.length -= 1
			return popped

	def unshift(self, value):
		if (self.length == 0):
			self.data[0] = value
		else:
			for i in range(self.length, 0, -1):
				self.data[i] = self.data[i-1]
			self.data[0] = value
		self.length += 1
		return self



mmmArray = myArray()
mmmArray.unshift("heyy")
print(mmmArray.printArray())
print(mmmArray.length)

File: Data Structures/array.js
class myArray{
	constructor(){
		this.length = 0
		this.data = {}
	}
	get(index){
		return this.data[index]
	}
	push(item){
		this.data[this.length] = item
		this.length++
		return this
	}
	pop(){
		if (this.length !== 0) {
			let popped = this.data[this.length-1]
			delete this.data[this.length-1]
			this.length--
			return popped
		} else {
			return "Nothing to pop"
		}
	}
	shift(){
		if (this.length !== 0) {
			let popped = this.data[0]
			let i = 0
			while(i < this.length) {
				this.data[i] = this.data[i+1]
				i++
			}
			delete this.data[this.length-1]
			this.length--
			return popped
		} else {
			return "Nothing to shift"
		}
	}
	unshift(item){
		
	}
	insert(index, item){

	}
	remove(index){

	}
	printArray(){
		return Object.values(this.data)
	}
}

let mymymymyArray = new myArray()
mymymymyArray.push(0)
mymymymyArray.push(10)
mymymymyArray.push(20)
console.log("shifted value : " , mymymymyArray.shift())
console.log("current array : " , mymymymyArray.printArray())
console.log(mymymymyArray)